### PR TITLE
CBCR-369: additional test routes for creationDate update and delete

### PR DESCRIPTION
**Pull request by brian-digital**

### CODE REVIEW

File: app/uk/gov/hmrc/cbcr/controllers/test/TestSubscriptionDataController.scala [hunk 1]
```diff
@@ -85,4 +88,27 @@ class TestSubscriptionDataController @Inject()(subRepo: SubscriptionDataReposito
...
   }
 
+  def updateReportingEntityCreationDate(docRefId:String, creationDate: String) = Action.async {
+    implicit request => {
+      val dri = DocRefId(docRefId)
+      val cd = LocalDate.parse(creationDate)
+
+      reportingEntityDataRepo.updateCreationDate(dri, cd).map {
+        case true => Ok
+        case false => NotModified
```
Comment (MaxWorgan) [inline app/uk/gov/hmrc/cbcr/controllers/test/TestSubscriptionDataController.scala]: update.ok doesn't say if anything has been modified, it just says nothing broke. Check out update.n which indicates the number of documents modified.

File: app/uk/gov/hmrc/cbcr/controllers/test/TestSubscriptionDataController.scala [hunk 2]
```diff
@@ -85,4 +88,27 @@ class TestSubscriptionDataController @Inject()(subRepo: SubscriptionDataReposito
...
+    }
+  }
+
+  def deleteReportingEntityCreationDate(docRefId:String) = Action.async {
+    implicit request => {
+      val dri = DocRefId(docRefId)
+
+      reportingEntityDataRepo.deleteCreationDate(dri).map {
+        case true => Ok
+        case false => NotModified
```
Comment (MaxWorgan) [inline app/uk/gov/hmrc/cbcr/controllers/test/TestSubscriptionDataController.scala]: as above

File: app/uk/gov/hmrc/cbcr/controllers/test/TestSubscriptionDataController.scala (hunk 1 above)
Comment (brian-digital) [reply app/uk/gov/hmrc/cbcr/controllers/test/TestSubscriptionDataController.scala]: now using update.nModified

File: app/uk/gov/hmrc/cbcr/controllers/test/TestSubscriptionDataController.scala (hunk 2 above)
Comment (brian-digital) [reply app/uk/gov/hmrc/cbcr/controllers/test/TestSubscriptionDataController.scala]: now using update.nModified

**Pull request merged**